pkg/apis/sources/v1beta1/apiserver_types.go

```diff
@@ -0,0 +1,149 @@
+	ApiServerSourceDeleteRefEventType = "dev.knative.apiserver.ref.delete"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ApiServerSourceList contains a list of ApiServerSource
+type ApiServerSourceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ApiServerSource `json:"items"`
+}
+
+// ApiServerSourceSpec defines the desired state of ApiServerSource
+type ApiServerSourceSpec struct {
+	// inherits duck/v1 SourceSpec, which currently provides:
+	// * Sink - a reference to an object that will resolve to a domain name or
+	//   a URI directly to use as the sink.
+	// * CloudEventOverrides - defines overrides to control the output format
+	//   and modifications of the event sent to the sink.
+	duckv1.SourceSpec `json:",inline"`
+
+	// WatchResources is the list of resources to watch.
+	WatchResources []ApiServerResource `json:"watch"`
+
+	// ServiceAccountName is the name of the ServiceAccount to use to run the
+	// receive adapter for this source.
+	// +optional
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+
+	// Mode is the mode the receive adapter controller runs under: Reference or Resource.
+	// `Reference` sends only the reference to the resource that had a lifecycle event.
+	// `Resource` send the full resource lifecycle event.
+	// Defaults to `Reference`
+	// +optional
+	Mode string `json:"mode,omitempty"`
+}
+
+// ApiServerSourceStatus defines the observed state of ApiServerSource
+type ApiServerSourceStatus struct {
+	// inherits duck/v1 SourceStatus, which currently provides:
+	// * ObservedGeneration - the 'Generation' of the Service that was last
+	//   processed by the controller.
+	// * Conditions - the latest available observations of a resource's current
+	//   state.
+	// * SinkURI - the current active sink URI that has been configured for the
+	//   Source.
+	duckv1.SourceStatus `json:",inline"`
+}
+
+// ApiServerSelectableKinds defines the resource to watch
+type ApiServerResource struct {
+	// API version of the resource to watch.
+	APIVersion string `json:"apiVersion"`
+
+	// Kind of the resource to watch.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	Kind string `json:"kind"`
+
+	// LabelSelector filters this source to objects to those resources pass the
+	// label selector.
+	// More info: http://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+	// +optional
+	LabelSelector *metav1.LabelSelector `json:"selector,omitempty"`
+
+	// +optional
+	ResourceOwner ApiServerResourceOwner `json:"owner,omitempty"`
+}
+
+type ApiServerResourceOwner struct {
+	// RestrictOwner restricts this source to objects with a controlling
+	// owner reference of the specified apiVersion and kind.
+	// +optional
+	RestrictOwner *APIVersionKind `json:"match,omitempty"`
```

Comment:
I think I'd hoist this up so it's at the same level as LabelSelector. At this point, I'm starting to think that perhaps I'm just really confused :) I'm sorry :(

Reply:
I was playing with the nesting context of `spec.owner`.

I liked:

```yaml
spec:
  owner:
    match:
      APIVersion: some.api/v1beta1
      Kind: FooBar
    watch: true
```

over

```yaml
spec:
  ownerMatch:
    APIVersion: some.api/v1beta1
    Kind: FooBar
  ownerWatch: true
```

Reply:
only because I think remembering (or autofilling) `ownerMatch` is harder than `owner` and then looking at what options are related to how to manipulate owner settings.